peerpod-ctrl/controllers/peerpod_controller.go

```diff
@@ -47,6 +59,20 @@ func (r *PeerPodReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	logger := log.FromContext(ctx)
 	pp := confidentialcontainersorgv1alpha1.PeerPod{}
 
+	// cloud provider was not set, try to fetch cloud provider and its configs dynamically from ConfigMap or Secret
+	// make sure the matching rbac rules are set
+	if r.Provider == nil {
+		if err := r.cloudConfigsGetter(); err != nil {
+			return ctrl.Result{RequeueAfter: (1 * time.Minute)}, nil
+		}
+
+		var pErr error
+		r.Provider, pErr = SetProvider()
+		if pErr != nil {
+			return ctrl.Result{}, pErr
+		}
+	}
+
```

Comment:
I suppose we don't need mutex here, since this method is not called concurrently. Right?

Reply:
setProvider? no